### ISSUE TITLE
examples/blob2d not working in next branch

### ISSUE THREAD

**Issue by JensMadsen** [boutproject/BOUT-dev#477]
from clean repo I get: 

```
Calculating differential geometry terms
Error encountered during initialisation: ====== Exception thrown ======
Field2D: (2, -1) index out of bounds (68 , 1)
====== Back trace ======
 -> Coordinates::geometry on line 217 of 'coordinates.cxx'
====== Exception path ======
[bt] #1 ./blob2d() [0x45b4d1]
BoutException::BoutException(char const*, ...) at /home/yolen/numerics/BOUT-dev/src/sys/boutexception.cxx:102 (discriminator 22)
[bt] #2 ./blob2d() [0x41421c]
Field2D::operator()(int, int) const at /home/yolen/numerics/BOUT-dev/src/field/../../include/field2d.hxx:181 (discriminator 2)
[bt] #3 ./blob2d() [0x468b5f]
Field2D::setYStencil(stencil&, bindex const&, CELL_LOC) const at /home/yolen/numerics/BOUT-dev/src/field/field2d.cxx:321
[bt] #4 ./blob2d() [0x44b56c]
Mesh::applyYdiff(Field2D const&, double (*)(stencil&), Mesh::boundary_derivs_pair (*)(forward_stencil&), Mesh::boundary_derivs_pair (*)(backward_stencil&), CELL_LOC) at /home/yolen/numerics/BOUT-dev/src/mesh/index_derivs.cxx:1172
[bt] #5 ./blob2d() [0x44b9d4]
Mesh::indexDDY(Field2D const&) at /home/yolen/numerics/BOUT-dev/src/mesh/index_derivs.cxx:1563
[bt] #6 ./blob2d() [0x43e5cd]
Coordinates::DDY(Field2D const&) at /home/yolen/numerics/BOUT-dev/src/mesh/coordinates.cxx:570
[bt] #7 ./blob2d() [0x43eb95]
Coordinates::geometry() at /home/yolen/numerics/BOUT-dev/src/mesh/coordinates.cxx:256 (discriminator 1)
[bt] #8 ./blob2d() [0x444dbf]
Coordinates::Coordinates(Mesh*) at /home/yolen/numerics/BOUT-dev/src/mesh/coordinates.cxx:155
[bt] #9 ./blob2d() [0x43427e]
Mesh::coordinates() at /home/yolen/numerics/BOUT-dev/src/mesh/mesh.cxx:263 (discriminator 2)
[bt] #10 ./blob2d() [0x4964e8]
BoutMesh::outputVars(Datafile&) at /home/yolen/numerics/BOUT-dev/src/mesh/impls/bout/boutmesh.cxx:2354
[bt] #11 ./blob2d() [0x467c83]
BoutInitialise(int&, char**&) at /home/yolen/numerics/BOUT-dev/src/bout++.cxx:306
[bt] #12 ./blob2d() [0x407e2a]
main at /home/yolen/numerics/BOUT-dev/examples/blob2d/../../include/boutmain.hxx:120
[bt] #13 /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf0) [0x7f4ab00bc830]
?? ??:0
[bt] #14 ./blob2d() [0x4074d9]
_start at ??:?

terminate called after throwing an instance of 'BoutException'
  what():  ====== Exception thrown ======
Field2D: (2, -1) index out of bounds (68 , 1)
====== Back trace ======
 -> Coordinates::geometry on line 217 of 'coordinates.cxx'
====== Exception path ======
[bt] #1 ./blob2d() [0x45b4d1]
BoutException::BoutException(char const*, ...) at /home/yolen/numerics/BOUT-dev/src/sys/boutexception.cxx:102 (discriminator 22)
[bt] #2 ./blob2d() [0x41421c]
Field2D::operator()(int, int) const at /home/yolen/numerics/BOUT-dev/src/field/../../include/field2d.hxx:181 (discriminator 2)
[bt] #3 ./blob2d() [0x468b5f]
Field2D::setYStencil(stencil&, bindex const&, CELL_LOC) const at /home/yolen/numerics/BOUT-dev/src/field/field2d.cxx:321
[bt] #4 ./blob2d() [0x44b56c]
Mesh::applyYdiff(Field2D const&, double (*)(stencil&), Mesh::boundary_derivs_pair (*)(forward_stencil&), Mesh::boundary_derivs_pair (*)(backward_stencil&), CELL_LOC) at /home/yolen/numerics/BOUT-dev/src/mesh/index_derivs.cxx:1172
[bt] #5 ./blob2d() [0x44b9d4]
Mesh::indexDDY(Field2D const&) at /home/yolen/numerics/BOUT-dev/src/mesh/index_derivs.cxx:1563
[bt] #6 ./blob2d() [0x43e5cd]
Coordinates::DDY(Field2D const&) at /home/yolen/numerics/BOUT-dev/src/mesh/coordinates.cxx:570
[bt] #7 ./blob2d() [0x43eb95]
Coordinates::geometry() at /home/yolen/numerics/BOUT-dev/src/mesh/coordinates.cxx:256 (discriminator 1)
[bt] #8 ./blob2d() [0x444dbf]
Coordinates::Coordinates(Mesh*) at /home/yolen/numerics/BOUT-dev/src/mesh/coordinates.cxx:155
[bt] #9 ./blob2d() [0x43427e]
Mesh::coordinates() at /home/yolen/numerics/BOUT-dev/src/mesh/mesh.cxx:263 (discriminator 2)
[bt] #10 ./blob2d() [0x4964e8]
BoutMesh::outputVars(Datafile&) at /home/yolen/numerics/BOUT-dev/src/mesh/impls/bout/boutmesh.cxx:2354
[bt] #11 ./blob2d() [0x467c83]
BoutInitialise(int&, char**&) at /home/yolen/numerics/BOUT-dev/src/bout++.cxx:306
[bt] #12 ./blob2d() [0x407e2a]
main at /home/yolen/numerics/BOUT-dev/examples/blob2d/../../include/boutmain.hxx:120
[bt] #13 /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf0) [0x7f4ab00bc830]
?? ??:0
[bt] #14 ./blob2d() [0x4074d9]
_start at ??:?

Aborted (core dumped)
```

Probably simple :-D

**Comment by JensMadsen** [comment]
Same error in several examples e.g. MMS/GBS

**Comment by dschwoerer** [comment]
Seems to me like a duplicate of #376

**Comment by bendudson** [comment]
This is now fixed by PR #485
